apps/mobile/src/screens/connection.tsx

```diff
@@ -44,16 +52,21 @@ function ActiveConnectionScreen({ connection }: { connection: Connection }) {
   }
 
   const onAction = useCallback(
-    (action: ActionDataState<string | string[]>) => {
-      const [name, path] = Array.isArray(action.name) ? action.name : [action.name, '']
+    (action: RiseAction) => {
+      const [name, payload] = Array.isArray(action.name) ? action.name : [action.name, '']
       if (name === 'navigate') {
-        router.push(`/connection/${connection.id}?path=${path}`)
+        router.push(`/connection/${connection.id}?path=${payload}`)
         return true
       }
       if (name === 'navigate-back') {
         router.back()
         return true
       }
+      if (action.name[0] === 'toast') {
+        const { title, message } = (action as ShowToastAction).name[1]
+        toast.show(title, { message })
+        return true
```

Comment:
Given this code, I propose the following API change to action:

Variant 1:
```ts
type ActionDataState<T> = { $: 'action', name: string, payload: T }
```

Variant 2
```ts
type ActionDataState<T> = { $: 'action', payload: T }
```

In Variant 1, we assume all actions within the system will need to have `name` and `payload` (optional).

In Variant 2, we just rename `name` to `payload` as I find it strange to access values from `name`. The fact that first value of payload is string, and second value(s) are action-specific data is something Rise-specific. Users can adopt a different shape within their projects.

While Variant 2 gives more flexibility, I think Variant 1 might provide a bit more interoperability across different projects, in case users decide to create reusable components that provide support for some actions. 

For example, there could be a 3rd party "ToastAction" component that users can drop-in within their apps to provide toast capability. Right now, it's hardcoded and spread out in many places within Rise, but there's nothing stopping us from turning it into reusable component, just like "DataBoundary".

Reply:
After discussing this with @ericvicenti on Discord, we're going with:
```
type ActionDataState<N, T> = { $: 'action', name: string } & T
```
This type might be actually broken, but you get the idea. Action "extends" from base object and it can provide some additional information.
We're changing name to be string.